src/utils/ticket/YouTrackTicketStatusFetcher.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace staabm\PHPStanTodoBy\utils\ticket;
+
+use RuntimeException;
+use staabm\PHPStanTodoBy\utils\CredentialsHelper;
+use staabm\PHPStanTodoBy\utils\HttpClient;
+use function array_key_exists;
+use function is_array;
+use function is_null;
+
+final class YouTrackTicketStatusFetcher implements TicketStatusFetcher
+{
+    private string $host;
+    private ?string $authorizationHeader;
+
+    private HttpClient $httpClient;
+
+    /**
+     * @var array<string, ?string>
+     */
+    private array $cache;
+
+    public function __construct(string $host, ?string $credentials, ?string $credentialsFilePath, HttpClient $httpClient)
+    {
+        $credentials = CredentialsHelper::getCredentials($credentials, $credentialsFilePath);
+
+        $this->host = $host;
+        $this->authorizationHeader = $credentials ? self::createAuthorizationHeader($credentials) : null;
+
+
+        $this->cache = [];
+        $this->httpClient = $httpClient;
+    }
+
+    public function fetchTicketStatus(string $ticketKey): ?string
+    {
+        if (array_key_exists($ticketKey, $this->cache)) {
+            return $this->cache[$ticketKey];
+        }
+
+        $url = "{$this->host}/api/issues/$ticketKey?fields=resolved";
+        $headers = [];
+        if (null !== $this->authorizationHeader) {
+            $headers = [
+                "Authorization: $this->authorizationHeader",
+            ];
+        }
+
+        [$responseCode, $response] = $this->httpClient->get($url, $headers);
+
+        if (200 !== $responseCode) {
+            throw new RuntimeException("Could not fetch ticket's status from Jira with url $url");
```

Comment:
```suggestion
            throw new RuntimeException("Could not fetch ticket's status from YouTrack with url $url");
```

Reply:
staabm also pointed this one out but thanks any way.